ThreadTimerTask - Java Timer Task which creates new long running Thread

jlhughes opened the pull request:
Java Timer Task which creates new long running thread for executing a ContextAgent.
Intended to be used by long running NanoServices.

jlhughes committed:
ThreadTimerTask - Java Timer Task which creates new long running
thread for executing a ContextAgent.
Intended to be used by long running NanoServices.

kgrgreer commented on src/foam/core/ThreadTimerTask.java:
```diff
@@ -0,0 +1,53 @@
+
+  public void run() {
+    Thread thread = new Thread(() -> {
+        X oldX = ((ProxyX) XLocator.get()).getX();
```
Does this work? Does the XLocator use the old or new thread? Should this be moved up two lines?

jlhughes replied:
Moved.

kgrgreer commented on src/foam/core/ThreadTimerTask.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.TimerTask;
+
+/**
+  TimerTask which executes ContextAgent in it's own thread.
```
it's -> its

kgrgreer commented on src/foam/core/ThreadTimerTask.java:
```diff
@@ -0,0 +1,53 @@
+  protected X x_;
+  protected ContextAgent agent_;
+  protected String name_;
+  public int priority_ = Thread.MAX_PRIORITY;
```
Do we want to do this? Could it starve other threads? Would default be sufficient?

jlhughes committed:
ThreadTimerTask - in run setup move acquisition of XLocator.X to
outside Thread creation block.
Add support to change priority and set default to NORMAL
